apps/web/src/utils/storage/companyDataStorage.ts

```diff
@@ -11,22 +11,30 @@ export default class CompanyDataStorage {
     [CompanyDataStorageKey.Size]: this.getSize()
   })
 
-  private static readonly _hasData: ComputedRef<boolean> = computed(() => {
-    return this._data.value[CompanyDataStorageKey.Company] !== null || this._data.value[CompanyDataStorageKey.Size] !== null
+  private static readonly _isDataFull: ComputedRef<boolean> = computed(() => {
+    return this._data.value[CompanyDataStorageKey.Company] !== null && this._data.value[CompanyDataStorageKey.Size] !== null
   })
 
   public static getData(): Ref<CompanyDataType> {
     return this._data
   }
 
-  public static hasData() {
-    return this._hasData
+  public static isDataFull() {
+    return this._isDataFull
   }
 
   public static hasCompanyData() {
     return this._data.value[CompanyDataStorageKey.Company] !== null
   }
 
+  public static hasSiret() {
+    if (!this._data.value[CompanyDataStorageKey.Company]) return false
+
+    return (
+      Object.hasOwn(this._data.value[CompanyDataStorageKey.Company], 'siret') &&
+      (this._data.value[CompanyDataStorageKey.Company] as EstablishmentFront).siret !== null
```

Comment:
suggestion: Est-ce que ceci fonctionnerais ?

```suggestion
return !!(this._data.value[CompanyDataStorageKey.Company] as EstablishmentFront)?.siret
```